File: 1876-substrings-of-size-three-with-distinct-characters/1876-substrings-of-size-three-with-distinct-characters.go
func countGoodSubstrings(s string) int {
    n := len(s)
    count := 0

    for i := 0; i+2 < n; i++ {
        if s[i] != s[i+1] &&
           s[i] != s[i+2] &&
           s[i+1] != s[i+2] {
            count++
        }
    }

    return count
}

File: 1876-substrings-of-size-three-with-distinct-characters/1876-substrings-of-size-three-with-distinct-characters.java
class Solution {
    public int countGoodSubstrings(String s) {
        int count = 0;
        for (int i = 2; i < s.length(); i++) {
            Character c1 = s.charAt(i - 2);
            Character c2 = s.charAt(i - 1);
            Character c3 = s.charAt(i);
            if (
                c1 != c2 && c2 != c3 && c3 != c1
            ) {
                count++;
            }
        }        
        return count;
    }
}